### PR TITLE
docs: proposed TRG 7.09 for repository deprecation

### LINKED ISSUE #1037: Propose TRG for product deprecation
https://github.com/orgs/eclipse-tractusx/security/overview

Ask committers to fix it, and give a deadline.

Take to the committer meeting.

---
Update Sep. 25 ⇾ regular topic in committer meeting ⇾ How much time do we give the committer to solve the problems? ⇒ New TRG proposal by Tom Meyer

---

Update Oct 2: Tom created https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/pull/1045, please review until next week

---

Update Oct 9: PR Review in Progess, will be discussed in Committer Meeting from Oct. 11 https://github.com/orgs/eclipse-tractusx/projects/61/views/6?pane=issue&itemId=79712658

### PR DESCRIPTION
**Pull request by tom-rm-meyer-ISST**
## Description

Proposed a TRG for product deprecation including a process and criteria to close #1037 

I didn't create a draft as I don't want to create two PRs following non-merged process within this [PR](https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/pull/871/files#diff-3210ef63113a43f664e14902d9cb192b6b77c3a7ac66b8b736db460fc7a5f14a).

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: docs/release/trg-7/trg-7-09.md
Comment (arnoweiss) [inline docs/release/trg-7/trg-7-09.md:15]:
```suggestion
Deprecation applies to repositories of type `product`. This information is maintained in the repository metafile (see [TRG 2.05](../trg-2/trg-2-5.md)). Deprecation of a repository means that **the repository is being archived**, i.e. set to status "Public Archive" on Github. It's still accessible from within the project, so that it could be revived. Please refer to [this list for examples of archived repositories within Eclipse Tractus-X](https://github.com/eclipse-tractusx/?q=&type=archived&language=&sort=).
```

**Comment by matbmoser** [comment]
@tom-rm-meyer-ISST should not the project leads deprecate the product? Not the committer it self. I would say the committer needs to inform the project leads

File: docs/release/trg-7/trg-7-09.md
Comment (evegufy) [inline docs/release/trg-7/trg-7-09.md:11]:
```suggestion
Eclipse Tractus-X is an open source project governed by the Eclipse Foundation and its Committers. Besides the role participating in maintaining the code base, committers are responsible for its security and adherence to the Tractus-X Releases Guideslines (TRG's) as well as the guidelines set by the Eclipse Foundation. As this group of committers is responsible for the whole project and has limited resources, repositories that are not maintained for a given time period will be deprecated.
```

File: docs/release/trg-7/trg-7-09.md
Comment (evegufy) [inline docs/release/trg-7/trg-7-09.md:2]: I'd write the TRG not specifically for Product Repositories but in general for Repositories
```suggestion
title: TRG 7.09 - Deprecation of Unmaintained Repositories
```

File: docs/release/trg-7/trg-7-09.md
Comment (evegufy) [inline docs/release/trg-7/trg-7-09.md:15]:
```suggestion
Deprecation of a repository means that **the repository is being archived**. It's still accessible from within the project, so that it could be revived. Please refer to [this list for examples of archived repositories within Eclipse Tractus-X](https://github.com/eclipse-tractusx/?q=&type=archived&language=&sort=).
```

File: docs/release/trg-7/trg-7-09.md
Comment (evegufy) [inline docs/release/trg-7/trg-7-09.md:24]:
```suggestion
1. The repository does not comply with the Tractus-X Releases Guidelines (TRG's) and/or the guidelines set by the Eclipse Foundation (i.e. Security, Intellectual Property, etc.)

In case of repositories belonging to a [Product](https://eclipse-tractusx.github.io/community/products):
2. The product of the repository has not been released within the *past year's major release cycle* **\***.
3. The standards used by the product have been changed in a major version within the *past year's major release cycle* **\***.
```

File: docs/release/trg-7/trg-7-09.md
Comment (evegufy) [inline docs/release/trg-7/trg-7-09.md:46]:
```suggestion
2. The committers inform the Eclipse Tractus-X community via the [mailing list](https://eclipse-tractusx.github.io/docs/oss/how-to-contribute/#dev-mailinglist). This is done by an issue with checkboxes listing all these repositories (find the issue template below). Additionally a critical issue is drafted on the [board for the committer meeting](https://github.com/orgs/eclipse-tractusx/projects/61/views/6) in at least 2 full weeks.
```

**Review by evegufy — CHANGES_REQUESTED** [review]
Hi @tom-rm-meyer-ISST thank you for proposing the TRG! Looks really good so far, just some comments.
If possible, I'd like to discuss the process in Committers Meeting this Friday (Oct. 11, 2024).

**Review by matbmoser — APPROVED** [review]

File: docs/release/trg-7/trg-7-09.md
Comment (matbmoser) [inline docs/release/trg-7/trg-7-09.md:42]:
```suggestion
    H -->|Yes| J[Plan Update of repository]
    J -->K[Create issue in .eclipsefdn]
```

**Review by matbmoser — CHANGES_REQUESTED** [review]

File: docs/release/trg-7/trg-7-09.md
Comment (arnoweiss) [inline docs/release/trg-7/trg-7-09.md:38]: Isn't this already covered by step 2?

Comment (tom-rm-meyer-ISST) [reply docs/release/trg-7/trg-7-09.md:41]: My idea was like: make the committers and the community aware of the issue. Wait for a grace period (at least one meeting in between) and then inform community "these are subject to deprecation"

I really want to emphasize, that this also somehow should be aligned best to the release cycle as it makes most sense to make persons prior to planning aware of this. One might also consider to trigger the evaluation always in the `.03` releases.

File: docs/release/trg-7/trg-7-09.md
Comment (tom-rm-meyer-ISST) [reply docs/release/trg-7/trg-7-09.md]: switched

File: docs/release/trg-7/trg-7-09.md
Comment (tom-rm-meyer-ISST) [reply docs/release/trg-7/trg-7-09.md]: done

File: docs/release/trg-7/trg-7-09.md
Comment (tom-rm-meyer-ISST) [reply docs/release/trg-7/trg-7-09.md]: done

**Comment by tom-rm-meyer-ISST** [comment]
Incorporated requested changes. Clearlydefined seems to be down.

File: docs/release/trg-7/trg-7-09.md
Comment (arnoweiss) [reply docs/release/trg-7/trg-7-09.md:41]: The problem with the dependency to releases (or open plannings) is that that'll extend the grace period to multiple months in the worst case. Do we want that?

File: docs/release/trg-7/trg-7-09.md
Comment (evegufy) [inline docs/release/trg-7/trg-7-09.md:82]:
```suggestion

**Supporting repository**:
- link to repository 3
```

File: docs/release/trg-7/trg-7-09.md
Comment (evegufy) [inline docs/release/trg-7/trg-7-09.md:83]:
```suggestion
**Call to Action**: Please discuss within your company e.g., with your Catena-X responsibles, whether you run into an issue by the deprecation. We highly encourage the companies to at least find resources for the **Network Service Products** to maintain an open source data space.
```

File: docs/release/trg-7/trg-7-09.md
Comment (evegufy) [inline docs/release/trg-7/trg-7-09.md:112]:
```suggestion
2. The requestor **MUST** announce the repository reactivation proposal on the [tractusx-dev](https://accounts.eclipse.org/mailing-list/tractusx-dev) mailing list mentioning the `issue Id`, the `reason/motivation`. A `responsible committer` **MUST** be announced.
```

File: docs/release/trg-7/trg-7-09.md
Comment (evegufy) [inline docs/release/trg-7/trg-7-09.md:94]:
```suggestion
1. The committer and requestor has capacity to resolve the security vulnerabilities and assure that the repository will comply and keep up to date TRGs.
```

File: docs/release/trg-7/trg-7-09.md
Comment (evegufy) [inline docs/release/trg-7/trg-7-09.md:73]:
```suggestion
Dear Eclipse Tractus-X Community,
```

File: docs/release/trg-7/trg-7-09.md
Comment (evegufy) [inline docs/release/trg-7/trg-7-09.md:49]:
```suggestion
1. The committers prepare a list of unmaintained repositories.
```

File: docs/release/trg-7/trg-7-09.md
Comment (evegufy) [inline docs/release/trg-7/trg-7-09.md:23]:
```suggestion
1. The repository does not comply with the Eclipse Tractus-X Releases Guidelines (TRG's) and/or the guidelines set by the Eclipse Foundation (i.e. Security, Intellectual Property, etc.)
```

File: docs/release/trg-7/trg-7-09.md
Comment (evegufy) [inline docs/release/trg-7/trg-7-09.md:42]:
```suggestion
```

File: docs/release/trg-7/trg-7-09.md
Comment (evegufy) [inline docs/release/trg-7/trg-7-09.md:51]:
```suggestion
3. 2 weeks period: The committers comment the issue regarding which repositories are confronted with the [deprecation criteria](#criteria-for-deprecation).
```

File: docs/release/trg-7/trg-7-09.md
Comment (evegufy) [inline docs/release/trg-7/trg-7-09.md:50]:
```suggestion
2. The committers inform the Eclipse Tractus-X community via the [dev mailing list](https://eclipse-tractusx.github.io/docs/oss/how-to-contribute/#dev-mailinglist). This is done by an issue with checkboxes listing all these repositories (find the issue template [below](#issue-template-for-deprecation-list)). Additionally a critical issue is drafted on the [board for the committer meeting](https://github.com/orgs/eclipse-tractusx/projects/61/views/6) in at least 2 full weeks.
```

File: docs/release/trg-7/trg-7-09.md
Comment (evegufy) [inline docs/release/trg-7/trg-7-09.md:55]:
```suggestion
5. If no action is taken within additional **4 weeks** the project leads will proceed with the deprecation of the given repositories and communicate accordingly on the dev mailing list.
6. In case the repository got deprecated, and there is again maintenance or develop capacity, the following procedure **MUST** be followed [Repository Reactivation Process](repository-reactivation-process) by a committer.
```

File: docs/release/trg-7/trg-7-09.md
Comment (evegufy) [inline docs/release/trg-7/trg-7-09.md:43]:
```suggestion
    G --> H{6: Action taken within 4 weeks?}
```

File: docs/release/trg-7/trg-7-09.md
Comment (evegufy) [inline docs/release/trg-7/trg-7-09.md:138]:
```suggestion
- [ ] I have emailed the mailing list announcing the committer assigned to the repository, providing a reasonable motivation for the repository reactivation.
```

**Review by evegufy — CHANGES_REQUESTED** [review]
sorry for the late re-review!

File: docs/release/trg-7/trg-7-09.md
Comment (tom-rm-meyer-ISST) [reply docs/release/trg-7/trg-7-09.md]: done

File: docs/release/trg-7/trg-7-09.md
Comment (tom-rm-meyer-ISST) [reply docs/release/trg-7/trg-7-09.md]: done

File: docs/release/trg-7/trg-7-09.md
Comment (tom-rm-meyer-ISST) [reply docs/release/trg-7/trg-7-09.md]: done

File: docs/release/trg-7/trg-7-09.md
Comment (tom-rm-meyer-ISST) [reply docs/release/trg-7/trg-7-09.md]: done but slightly different

File: docs/release/trg-7/trg-7-09.md
Comment (tom-rm-meyer-ISST) [reply docs/release/trg-7/trg-7-09.md]: done

File: docs/release/trg-7/trg-7-09.md
Comment (tom-rm-meyer-ISST) [reply docs/release/trg-7/trg-7-09.md]: done

File: docs/release/trg-7/trg-7-09.md
Comment (tom-rm-meyer-ISST) [reply docs/release/trg-7/trg-7-09.md]: done

File: docs/release/trg-7/trg-7-09.md
Comment (tom-rm-meyer-ISST) [reply docs/release/trg-7/trg-7-09.md]: checked spelling of fulfill. It's really with two l. Thanks for paying that much attention on typos!

File: docs/release/trg-7/trg-7-09.md
Comment (tom-rm-meyer-ISST) [reply docs/release/trg-7/trg-7-09.md]: done

File: docs/release/trg-7/trg-7-09.md
Comment (tom-rm-meyer-ISST) [reply docs/release/trg-7/trg-7-09.md]: done

File: docs/release/trg-7/trg-7-09.md
Comment (tom-rm-meyer-ISST) [reply docs/release/trg-7/trg-7-09.md]: done

File: docs/release/trg-7/trg-7-09.md
Comment (tom-rm-meyer-ISST) [reply docs/release/trg-7/trg-7-09.md]: done

**Review by evegufy — APPROVED** [review]
Thank you @tom-rm-meyer-ISST

**Comment by tom-rm-meyer-ISST** [comment]
@arnoweiss & @matbmoser please check. I think three reviews would be sufficient

**Review by arnoweiss — APPROVED** [review]

**Review by matbmoser — APPROVED** [review]
LGTM

**Comment by matbmoser** [comment]
Wonderful job @tom-rm-meyer-ISST !

**Comment by tom-rm-meyer-ISST** [comment]
@evegufy do you want to cross-check again? Else I would merge latest eob and write a short mail to the dev mailing list. 

I won't be available in todays meeting (that's why I'm writing here :))

**Comment by evegufy** [comment]
@tom-rm-meyer-ISST all fine from my side. Yes, please inform over the mailing list once merged, thank you :)

**Pull request merged**
